### ISSUE TITLE
IAM Role is accessible by anyone

### ISSUE THREAD

**Issue by RyanJarv** [terraform-community-modules/tf_aws_nat#25]
Looks like the IAM role this module creates is accessible by anyone:

https://github.com/terraform-community-modules/tf_aws_nat/blob/3c72f510cf751c90eb70cd8e840b709bb62b9b74/iam.tf#L7C10-L7C35

You likely want the principal to be something like this if your using it with EC2:

```
            "Principal": {
                "Service": [
                    "ec2.amazonaws.com"
                ]
            }
```

**Comment by antonbabenko** [comment]
This module is enormously outdated and won't be updated in the future.

**Comment by christophetd** [comment]
Might be a good idea to say so in the README, then?

**Comment by antonbabenko** [comment]
Fixed in #20, released tag `v1.5.0`, and updated a repo description.

I am surprised that the Terraform module written 9 years ago is still in use and even working for some people :)

**Comment by RyanJarv** [comment]
You may be right, I just found this when doing some digging on sourcegraph.com for some unrelated stuff. This had already been featured in an example repo for insecure code (think it was test code for a terraform security scanner? I don't remember now) and was a bit surprised to see it hadn't been fixed or deprecated yet.

Glad to see it's resolved now!
